src/util/gc_byte.rs

```diff
@@ -1,11 +1,35 @@
-use crate::util::ObjectReference;
+use crate::util::side_metadata::{SideMetadata, SideMetadataID};
+use crate::util::{constants, ObjectReference};
 use crate::vm::ObjectModel;
 use crate::vm::VMBinding;
 use std::sync::atomic::{AtomicU8, Ordering};
 
+use super::Address;
+
 /// This struct encapsulates operations on the per-object GC byte (metadata)
 pub struct GCByte {}
 
+// For performance reasons, mutable static is used over lazy_static or sync::once.
+// This requires several unsafe blocks in the current file, but it looks safe to use these unsafe blocks, because we only assign the mutable static once.
+//
+// NOTE: A more suitable implementation may also be possible.
+static mut SIDE_GCBYTE_ID: SideMetadataID = SideMetadataID::new();
+
+#[allow(clippy::cast_ref_to_mut)]
+#[allow(clippy::mut_from_ref)]
+pub(crate) unsafe fn init_side_gcbyte() {
+    let res = SideMetadata::request_meta_bits(
```

Comment:
I think it makes more sense if we reserve a whole byte instead of 2 bits for each word. Otherwise this should be called "gc byte" since it is not a whole byte.

Reply:
I remember we had a discussion about this, and the conclusion is to use 2 bits to save space. Please correct me if I am wrong.

Reply:
It's called side GC-byte in the sense that it is an alternative to the in-header GC-byte, and its interface uses the same (`u8`) type, and it can be extended to 8-bits if the selected plan wants.
(I'm replacing the constant `2` with `SelectedConstraints::GC_HEADER_BITS` in my next commit to this PR).
In fact, no one except the internal implementation of the GC-byte module, will know about the 2 or 8-bits.